### PR TITLE
Create june-21.md

### PR DESCRIPTION
**Pull request by hzoo**

### CODE REVIEW

**Comment by nathanhammond** [comment]
I've added the notes as they were during the meeting, still need to clean them up for clarity.

**Comment by sarupbanskota** [comment]
there's also https://github.com/babel/babel/issues/5878 related to plugin ordering

**Comment by peey** [comment]
I don't have write access. Some minor edits:

```
### Inline Parameters

- https://github.com/babel/babel/pull/5751 by @peey
- Moves the parameters into the function body so that the normal rest spread can be reused.
- Not a separate visitor, instead it's a method on the path object `fnPath.inlineParameters()`
- Object Rest Spread had bugs because logic was duplicated in paramters transform.
- Counts as a refactor of Henry's code.
- Request of additional documentation.
- Handles default parameters (including spread), rest, ObjectPattern and ArrayPattern
```

File: 2017-06/june-21.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+- Handles default parameters (including spread), rest, ObjectPattern and ArrayPattern
+
+### TypeScript in Babylon
+
+- Andy's PRs: https://github.com/babel/babylon/pull/523, https://github.com/babel/babel/pull/5856
+- Lots and lots of code, it's kind of a patch bomb.
+- Doesn't appear to be many changes to core Babylon.
+- Most of the code is in the plugin.
+- If the core changes make sense, and are reasonable, we can defer to Microsoft for maintenance of anything they've upstreamed.
+- Is there any particular reason for us to include the plugin in core?
```
Comment (Kovensky) [inline 2017-06/june-21.md]: Using babel to transform typescript into typescript?

**Pull request merged**